### ISSUE TITLE
📎 Implement `useFocusableInteractive` - `jsx-a11y/interactive-supports-focus`

### ISSUE THREAD

**Issue by Conaclos** [biomejs/biome#2611]
### Description

Implement [interactive-supports-focus](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/interactive-supports-focus.md).

Want to contribute? Lets you know you are interested! We will assign you to the issue to prevent several people to work on the same issue. Don't worry, we can unassign you later if you are no longer interested in the issue! Read our [contributing guide](https://github.com/biomejs/biome/blob/main/CONTRIBUTING.md) and [analyzer contributing guide](https://github.com/biomejs/biome/blob/main/crates/biome_analyze/CONTRIBUTING.md).

Suggested name: `useFocusableInteractive`

**Comment by fujiyamaorange** [comment]
@Conaclos 
The eslint link looks same as https://github.com/biomejs/biome/issues/2610 👀

**Comment by DerTimonius** [comment]
I'd like to work on this one 🙂

**Comment by DerTimonius** [comment]
Currently working on this, I'm already making good progress!

There are two questions that came up:

1. Should the third case of the original rule be included here? To me, it feels a bit disconnected
 >Case: This element is not a button, link, menuitem, etc. It is catching bubbled events from elements that it contains
If your element is catching bubbled click or key events from descendant elements, then the proper role for this element is presentation.

```jsx
<div
  onClick={onClickHandler}
  role="presentation">
  <button>Save</button>
</div> 
```
> Marking an element with the role presentation indicates to assistive technology that this element should be ignored; it exists to support the web application and is not meant for humans to interact with directly.

2. Should the rule have options to make some elements not only focusable, but also tabbable as suggested in the original rule?

**Comment by ematipico** [comment]
1. Yeah you're right, the third case is weird. In that case, we should suggest a different role (covered by another rule I think). We should ignore it.
2. Usually we start without options unless they are needed for the first round of implementation.
